### PR TITLE
feat(test): rbac realm tests

### PR DESCRIPTION
**Pull request by jinoosss**
## Descriptions

This PR adds comprehensive test coverage for RBAC (Role-Based Access Control) functionality at the realm level. The tests cover all major RBAC operations including role registration, permission management, and access control validation.

#### RBAC Manager Tests (`manager_test.gno`)
- **`TestRegisterRole`**: Tests role registration with success/failure scenarios
- **`TestRegisterPermission`**: Tests permission registration for roles
- **`TestCheckPermission`**: Tests permission validation for different callers
- **`TestUpdatePermission`**: Tests permission checker updates
- **`TestRemovePermission`**: Tests permission removal functionality
- **`TestRemoveRole`**: Tests role deletion with authorization checks
- **`TestDeclareRole`**: Tests declarative role creation with permissions

### Authorization & Security
- Admin-only operations validation
- Unauthorized access prevention
- Role ownership verification
- Permission checker validation

### Error Handling
- Non-existent role/permission handling
- Duplicate registration prevention
- Invalid caller address validation
- System role protection (admin role cannot be deleted)

### Realm Integration
- Cross-realm permission checking
- User realm vs code realm validation
- Multi-role address scenarios
- Dynamic role management

### Success Cases
- Role registration with valid parameters
- Permission assignment to existing roles
- Permission validation for authorized users
- Role updates by admin users
- Dynamic role creation and management

### Failure Cases
- Unauthorized access attempts
- Non-existent role/permission access
- Duplicate role registration
- Invalid caller address validation
- System role deletion attempts

### Improved Security
- Comprehensive validation of RBAC operations
- Proper authorization checks for all sensitive operations
- Protection against unauthorized role modifications

### Better Maintainability
- Clear test coverage for all RBAC functionality
- Documented expected behavior for edge cases
- Regression prevention for future changes

### Enhanced Reliability
- Validation of cross-realm interactions
- Testing of permission checker functions
- Verification of role lifecycle management

## Test Statistics

- **Total Test Functions**: 10+
- **Test Cases**: 50+ individual scenarios
- **Coverage Areas**: Role management, permission handling, access control
- **Error Scenarios**: 20+ failure case validations

### Test Patterns
- Table-driven tests for comprehensive scenario coverage
- Given-When-Then structure for clear test organization
- Proper setup and teardown for isolated test execution
- Cross-realm testing for realistic scenarios

### CODE REVIEW

**Review by notJoon — APPROVED** [review]
LGTM :+1

File: contract/r/gnoswap/rbac/manager_test.gno [hunk 1]
```diff
@@ -7,145 +7,542 @@ import (
...
+			inputPermissionName: "can_delete",
+			inputChecker: func(addr std.Address) error {
+				if addr == testCaller {
+					return nil
+				}
+				return errors.New("new checker failed")
+			},
+			expectedError:      nil,
+			expectedCheckError: errors.New("new checker failed"),
+		},
```
Comment (jinoosss) [inline contract/r/gnoswap/rbac/manager_test.gno:260]: @dongwon8247 ,
We are checking whether it has been changed from the default `checker` function to the new `checker` function.

If the checker function change is successful, you can see the expected `new checker failed` error in the test.

**Review by dongwon8247 — APPROVED** [review]
